mne_bids/path.py

```diff
@@ -120,22 +132,88 @@ def __str__(self):
                     key = 'rec'
                 basename.append('%s-%s' % (key, val))
 
-        if self.suffix is not None:
-            basename.append(self.suffix)
+        if self.kind is not None:
+            if self.extension is not None:
+                basename.append(f'{self.kind}{self.extension}')
+            else:
+                basename.append(self.kind)
 
         basename = '_'.join(basename)
-        if self.prefix is not None:
-            basename = op.join(self.prefix, basename)
-
         return basename
 
+    @property
+    def fpath(self):
+        """Get the full filepath for this BIDS file.
+
+        Getting the file path consists of the following behavior:
+
+        1. ``bids_root`` is not passed into the object: A warning is shown
+        to the user and it return the ``basename``.
+
+        2. ``bids_root`` is passed into the object:
+
+            i. If ``kind`` and ``extension`` are given:
+            The function will then return the full filepath
+            using ``bids_root`` and ``basename``
+
+            ii. Else:
+            The function tries to infer the full filepath
+            for the given ``bids_root`` and ``basename``
+            and will return the full filepath. It will try to
+            infer the filename extension by searching
+            for the file on disk. If the file cannot be found, an error
+            will be raised. To disable this automatic inference attempt,
+            pass a string (like ``'.fif'`` or ``'.vhdr'``).
+
+            If it cannot find a matching dataset, it
+            will result in an error.
+
+        If ``None``,
```

Comment:
I think this comment is kind of superfluous. It's basically writing out in English what you have written below with if-else. So the value is kind of low

Reply:
maybe better if you can condense it and put it in the description of `bids_fpath` below

Reply:
I added it in. lmk what you think.